sqlglot/parser.py

```diff
@@ -1352,6 +1372,14 @@ def _parse_types(self):
             nested=nested,
         )
 
+    def _parse_struct_kwargs(self):
+        this = self._parse_id_var()
+        self._match(TokenType.COLON)
+        data_type = self._parse_types()
+        if not data_type:
+            return None
+        return self.expression(exp.StructKwarg, this=this, expression=data_type)
```

Comment:
Yeah this is much cleaner. I hadn't thought of the option of taking the Struct type and making it into an expression.